packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -872,18 +875,27 @@ class ScrollableState extends State<Scrollable>
     assert(_drag == null);
     _drag = position.drag(details, _disposeDrag);
     assert(_drag != null);
-    assert(_hold == null);
+    // _hold might be non-null if the scroll position is currently animating.
+    if (_hold != null) {
+      _disposeHold();
+    }
   }
 
   void _handleDragUpdate(DragUpdateDetails details) {
+    // _hold might be non-null if the scroll position is currently animating.
+    if (_hold != null) {
+      _disposeHold();
+    }
     // _drag might be null if the drag activity ended and called _disposeDrag.
-    assert(_hold == null || _drag == null);
```

Comment:
Why remove this assertion? Do we have a crash here when the one in `_handleDragStart` is addressed? This assertion is not just about the _hold, which gives me some pause here.